apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,563 @@
+
+### CJS source and only ESM distribution
+
+The "Gin & Tonic" of packages: This takes a small bit of finesse but is also pretty straight-forward.
+
+**Working example**: [cjs-with-esm-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/esm-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-source-with-esm-distribution",
+  "main": "./dist/index.mjs"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-source-with-esm-distribution",
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/index.mjs",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+The [`.mjs`](https://nodejs.org/api/esm.html#enabling) file extension is a trump-card: it will override **any** other configuration and the file will be treated as ESM. Using this file extension is necessary because `packageJson.exports.import` does **NOT** signify that the file is ESM (contrary to common, if not universal, misperception), only that it is the file to be used when the package is imported (ESM _can_ import CJS. See [Gotchas](#gotchas) below).
+
+### CJS source and both CJS & ESM distribution
+
+You have a few options:
+
+#### Attach named exports directly onto `exports`
+
+The "French 75" of packages: Classic but takes some sophistication and finesse.
+
+Pros:
+
+- Smaller package weight
+- Easy and simple (probably least effort if you don't mind keeping to a minor syntax stipulation)
+- Precludes the Dual-Package Hazard
+
+Cons:
+
+- Requires very specific syntax (either in source code and/or bundler gymnastics).
+
+**Working example**: [cjs-with-dual-distro (properties)](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/dual/property-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-source-with-esm-via-properties-distribution",
+  "main": "./dist/cjs/index.js"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-source-with-esm-via-properties-distribution",
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/cjs/index.js",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+Typically, you would see `module.exports` assigned to something (be it an object or a function) like this:
+
+```cjs
+const someObject = {
+  foo() {},
+  bar() {},
+  qux() {},
+};
+
+module.exports = someObject;
+```
+
+Instead, do this:
+
+```cjs
+module.exports.foo = function foo() {};
+module.exports.foo = function bar() {};
+module.exports.foo = function qux() {};
+```
+
+#### Use a simple ESM wrapper
+
+The "Piña Colada" of packages: Complicated setup and difficult to get the balance right.
+
+Pros:
+
+- Smaller package weight
+
+Cons:
+
+- Likely requires complicated bundler gymnastics (we could not find any existing option to automate this in Webpack).
```

Comment:
> Also an obvious cons is that one needs to manually keep the exports in-sync.

That's precisely the complicated bundler gymnastics

Reply:
```suggestion
- Likely requires complicated bundler gymnastics to keep exports in-sync (we could not find any existing option to automate this in Webpack).
```

Reply:
```suggestion
- Requires manually maintained code to keep exports in-sync.
```

I think this should be enough, at least to me I still don't know how this can be interpreted as `complicated bundler gymnastics`. This pattern is only relevant when the bundlers can't generate detectable code, so the "Likely" should be removed (it's a precondition). Also in this case the user no longer needs to worry about the patterns emitted by a bundler anymore because they are doing the chores themselves, which means more like "no bundler gymnastics required" to me.